examples/workflows/callable_dag_with_param_get.py

```diff
@@ -0,0 +1,31 @@
+from typing_extensions import Annotated
+
+from hera.workflows import DAG, Parameter, Workflow, script
+
+
+@script(constructor="runner")
+def hello_with_output(name: str) -> Annotated[str, Parameter(name="output-message")]:
+    return "Hello, {name}!".format(name=name)
+
+
+with Workflow(
+    generate_name="callable-dag-",
+    entrypoint="calling-dag",
+) as w:
+    with DAG(
+        name="my-dag-with-outputs",
+        inputs=Parameter(name="my-dag-input"),
+        outputs=Parameter(
+            name="my-dag-output",
+            value_from={
+                "parameter": "{{hello.outputs.parameters.output-message}}"
+            },  # Don't think we can improve this?
+        ),
+    ) as my_dag:
+        hello_with_output(name="hello", arguments={"name": f"hello {my_dag.get_parameter('my-dag-input')}"})
```

Comment:
Depending on if we're keeping the explanation comments of get_parameter, would suggest:

```suggestion
        # Here, get_parameter searches through the *inputs* of my_dag
        hello_with_output(name="hello", arguments={"name": f"hello {my_dag.get_parameter('my-dag-input')}"})
```